src/Bicep.LangServer/Handlers/BicepDecompileForPasteCommandHandler.cs

```diff
@@ -80,22 +86,47 @@ public override Task<BicepDecompileForPasteCommandResult> Handle(BicepDecompileF
             });
         }
 
+        private string DisclaimerMessage => $"{LangServerResources.DecompileAsPaste_AutoConvertWarning}\n{BicepDecompiler.DecompilerDisclaimerMessage}";
+
+        private static void Log(StringBuilder output, string message)
+        {
+            output.AppendLine(message);
+            Trace.TraceInformation(message);
+        }
+
         private async Task<ResultAndTelemetry> TryDecompileForPaste(string json, bool queryCanPaste)
         {
             StringBuilder output = new StringBuilder();
             string decompileId = Guid.NewGuid().ToString();
 
-            var (pasteType, constructedJsonTemplate, needsDisclaimer) = ConstructFullJsonTemplate(json);
+            var (pasteType, constructedJsonTemplate) = TryConstructFullJsonTemplate(json);
             if (pasteType is null)
             {
+                // It's not a template or resource.  Try treating it as a JSON value.
+                var resultAndTelemetry = TryConvertFromJsonValue(output, json, decompileId, queryCanPaste);
+                if (resultAndTelemetry is not null)
+                {
+                    Log(output, DisclaimerMessage);
```

Comment:
Is there a reason why we are not logging anything to this channel - Bicep Operations?

Reply:
Bug.  Also cleaned up and improved behavior.

f5dd472e3